### DISCUSSION TITLE
How to display a list in descending order

### DISCUSSION THREAD

**Discussion by mg72266** [axuno/SmartFormat#333] [Q&A] [answered]
Is there a way to display a list in descending order?

Given an array,

`var array = new List<string> { "555100", "555101", "555102" };`

I have a string that I format it like so,

`var string = Smart.Format("array - {0:{:substr(3)}| }", array)`

Am I able to format the string so I can have the array appear in descending order without reversing the array object?

**Comment by axunonb** [comment, marked as answer]
There is no parameter or setting for `ListFormatter` to process items in reverse order.
This can be achieve with a custom implementation of the `ListFormatter` (e.g. `RListFormatter`, although this looks like an overkill.

**Reply by mg72266** [reply]
Thanks for your reply and suggestion. I was able to implement a reverse function by implementing ISource and looking at how TryEvaluateSelector was implemented in ListSource as a guide.

**Reply by axunonb** [reply]
Excellent, congrats!
